torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -336,6 +336,12 @@ def __init__(
             if submodule not in self._ignored_modules:
                 submodule._is_fsdp_managed_module = True
 
+                # Dynamo only supports FSDP with use_orig_params=True.
+                # This is hacky, but I could not think of another way to add an assertion to dynamo
+                # for this, since Dynamo skips all the FSDP code frames and thus can't inspect the
+                # FSDP module directly
+                submodule._fsdp_use_orig_params = use_orig_params
+
```

Comment:
This FSDP change looks fine to me!

Reply:
ok. the rest of the change is pretty trivial. I think the biggest risk is that we are cluttering up the submodules this way.  the dynamo change is not a risk, and the unit test works.